atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/BackgroundSweeperFastTest.java

```diff
@@ -224,6 +232,19 @@ public void testMetricsUseIntermediateResultsPerIteration() {
 
         setupTaskRunner(intermediateResults);
         backgroundSweeper.runOnce();
-        Mockito.verify(sweepMetricsManager).updateMetrics(intermediateResults, TABLE_REF);
+
+        ArgumentCaptor<Long> sweepTime = ArgumentCaptor.forClass(Long.class);
+        ArgumentCaptor<Long> totalTimeElapsed = ArgumentCaptor.forClass(Long.class);
+        Mockito.verify(sweepMetrics).updateSweepTime(
+                sweepTime.capture(),
+                totalTimeElapsed.capture());
+
+        Assertions.assertThat(intermediateResults.getTimeInMillis()).isEqualTo(sweepTime.getValue());
+        assertWithinErrorMarginOf(intermediateResults.getTimeElapsedSinceStartedSweeping(), totalTimeElapsed.getValue());
+    }
+
+    private void assertWithinErrorMarginOf(long actual, long expected) {
+        Assertions.assertThat(actual).isGreaterThanOrEqualTo((long) (expected * .95));
+        Assertions.assertThat(actual).isLessThanOrEqualTo((long) (expected * 1.05));
```

Comment:
nit: This is equivalent to `assertThat(actual).isCloseTo(expected, withinPercentage(5d));`, and depends on taste but I would probably inline the calls as it's a one-liner after that

Reply:
nice!